seed/csharp-model/enum/forward-compatible-enums/src/SeedEnum/Color.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Text.Json.Serialization;
+using SeedEnum.Core;
+
+#nullable enable
+
+namespace SeedEnum;
+
+[JsonConverter(typeof(StringEnumSerializer<Color>))]
+public readonly struct Color : IStringEnum, IEquatable<Color>
```

Comment:
would using a `record struct` simplify any of the equality logic?

Reply:
I'll take a look at that and verify if its backwards compatible to older .NET versions with PolySharp, that would be great

Reply:
sounds good, you may already be aware but we use `record`s for our generated object types and `dotnet build` doesn't complain

(my understanding is that the command will complain if anything we write is not compatible with any of our target frameworks, which happened when i tried to use `record` without PolySharp)

Reply:
There is potential for some of these features to not work at runtime, but running tests in the runtimes we support should remove any doubt.